tests/test_datasets/test_dataset_functions.py

```diff
@@ -261,46 +223,40 @@ def test__name_to_id_version_does_not_exist(self):
     def test_get_datasets_by_name(self):
         # did 1 and 2 on the test server:
         dids = ["anneal", "kr-vs-kp"]
-        datasets = openml.datasets.get_datasets(dids, download_data=False)
+        datasets = openml.datasets.get_datasets(dids)
         assert len(datasets) == 2
-        self._datasets_retrieved_successfully([1, 2])
+        _assert_datasets_retrieved_successfully([1, 2])
 
     def test_get_datasets_by_mixed(self):
         # did 1 and 2 on the test server:
         dids = ["anneal", 2]
-        datasets = openml.datasets.get_datasets(dids, download_data=False)
+        datasets = openml.datasets.get_datasets(dids)
         assert len(datasets) == 2
-        self._datasets_retrieved_successfully([1, 2])
+        _assert_datasets_retrieved_successfully([1, 2])
 
     def test_get_datasets(self):
         dids = [1, 2]
         datasets = openml.datasets.get_datasets(dids)
         assert len(datasets) == 2
-        self._datasets_retrieved_successfully([1, 2], metadata_only=False)
-
-    def test_get_datasets_lazy(self):
-        dids = [1, 2]
-        datasets = openml.datasets.get_datasets(dids, download_data=False)
-        assert len(datasets) == 2
-        self._datasets_retrieved_successfully([1, 2], metadata_only=True)
+        _assert_datasets_retrieved_successfully([1, 2])
 
-        datasets[0].get_data()
-        datasets[1].get_data()
-        self._datasets_retrieved_successfully([1, 2], metadata_only=False)
-
-    @pytest.mark.production()
     def test_get_dataset_by_name(self):
         dataset = openml.datasets.get_dataset("anneal")
         assert type(dataset) == OpenMLDataset
         assert dataset.dataset_id == 1
-        self._datasets_retrieved_successfully([1], metadata_only=False)
+        _assert_datasets_retrieved_successfully([1])
 
         assert len(dataset.features) > 1
         assert len(dataset.qualities) > 4
 
-        # Issue324 Properly handle private datasets when trying to access them
-        openml.config.server = self.production_server
-        self.assertRaises(OpenMLPrivateDatasetError, openml.datasets.get_dataset, 45)
```

Comment:
This was duplicated in multiple tests. I extracted it as its own test. Does not seem to depend on whatever came before.